### DISCUSSION TITLE
Defer validation until submit is clicked

### DISCUSSION THREAD

**Discussion by cill-i-am** [airjp73/rvf#108] [Q&A] [answered]
I'm building a form using zod as a validator, everything is going great! But I have a requirement for the error labels to not display until submit is clicked. Is there any way to defer clientside validation until then? Or even to not validate the form if the fields are empty? 

My issue is that if I autofocus on an email input and the user clicks away without entering anything, then the validation triggers

**Comment by airjp73** [comment, marked as answer]
Hi! Sorry for the slow answer. You're looking for the [validationBehavior](https://www.remix-validated-form.io/reference/use-field#validationBehavior) option on `useField`. Usage would be like this:

```tsx
useField(fieldName, {
  validationBehavior: {
    // Set initial and whenTouched to "onSubmit" and it won't validate until submit
    initial: "onSubmit",
    whenTouched: "onSubmit",

    // If you want this behavior to change after submit, you can change that here
    whenSubmitted: "onChange"
  }
})
```

**Reply by cill-i-am** [reply]
This is perfect, thank you so much 👍

**Reply by kwiat1990** [reply]
Can we defined default options for `validationBehavior `globally, so that it doesn't need to be passed in to every `useField` instance separately?
